website/blog/2018-07-14-whats-the-deal-with-the-new-pipeline-plugin-configuration.md

```diff
@@ -0,0 +1,153 @@
+---
+layout: post
+title:  "What's the Deal with the New Pipeline Plugin Configuration?"
+author: James DiGioia
+authorURL: https://twitter.com/JamesDiGioia
+date:   2018-06-26 12:00:00
+categories: announcements
+share_text: "What's the Deal with the New Pipeline Plugin Configuration?"
+---
+
+With the release of babel@7.0.0-beta52, we introduced a new required configuration flag to `@babel/plugin-proposal-pipeline-operator`, a breaking change not only for the pipeline operator itself but for the `stage-0` & `stage-1` presets that use it. In order to clear up some of the confusion, let's take a look at the pipeline proposal and why we needed to introduce this configuration option.
+
+<!--truncate-->
+
+## Current Status
+
+The pipeline operator was originally introduced by [Gilbert Garza](https://github.com/gilbert), providing a clean syntax for "streamlining chained function calls in a readable, functional manner." The pipeline operator has roots in a number of languages, including F#, Hack, Elm, Elixir, and others, but there were two major points of contention in introducing the new syntax to JavaScript:
+
+* Whether and how to introduce placeholders
+* How to handle async / await in the pipeline
+
+### Placeholders
+
+The first issue was the question of placeholders. This was first raised by [Kevin Smith](https://github.com/zenparsing) in [this issue](https://github.com/tc39/proposal-pipeline-operator/issues/84), suggesting Hack-style pipelining requring placeholders for every function call. Placeholders expand the potential use space of the pipeline operator, as you can now embed arbitrary expressions in the pipeline, with the placeholder representing the value returned from the previous step.
+
+The downside is the complexity involved in introducing a new token. The hash (`#`) is the current choice, and although this is still open to bikeshedding, any token would potentially have multiple meanings. The hash is also used by the private fields proposal, and [all other options are in use in one form or another](https://github.com/tc39/proposal-partial-application/issues/21#issuecomment-361092565).
+
+### Async / Await
+
+The initial introduction of the pipeline included this syntax for `await`:
+
+```js
+x |> await f
+```
+
+which would desugar to
+
+```js
+await f(x)
+```
+
+Unfortunately, this is intuitively ambiguous, as it could reasonably desugar to this:
+
+```js
+(await f)(x)
+```
+
+While there was some pushback on the idea of including async handling in the pipeline at all, the committee felt, without a strong reason it wasn't doable, it would oppose a pipeline operator that didn't handle async/await. While there are ways to handle Promise-returning functions without explicit syntax, they are too cumbersome to be useful or require a helper function.
```

Comment:
It is possible to remove some of the extra commas here?

While there was some pushback on the idea of including async handling in the pipeline at all, the committee still chose to oppose a pipeline operator that didn't handle async/await?